mmv1/products/firebaseappcheck/DeviceCheckConfig.yaml

```diff
@@ -0,0 +1,113 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "DeviceCheckConfig"
+base_url: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig
+self_link: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig
+create_url: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig?updateMask=tokenTtl,keyId,privateKey
+create_verb: :PATCH
+update_verb: :PATCH
+update_mask: true
+skip_delete: true
+description: |
+  An app's DeviceCheck configuration object. Note that the Team ID registered with your
+  app is used as part of the validation process. Make sure your `google_firebase_apple_app` has a team_id present.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://firebase.google.com/docs/app-check"
+  api: "https://firebase.google.com/docs/reference/appcheck/rest/v1/projects.apps.deviceCheckConfig"
+import_format:
+  [
+    "projects/{{project}}/apps/{{app_id}}/deviceCheckConfig",
+    "{{project}}/{{app_id}}",
+    "{{app_id}}",
+  ]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebase_app_check_device_check_config_minimal"
+    # Need the time_sleep resource
+    pull_external: true
+    primary_resource_id: "default"
+    vars:
+      bundle_id: "bundle.id.devicecheck"
+      key_id: "Key ID"
+      private_key_path: "path/to/private-key.p8"
+      team_id: "9987654321"
+    test_vars_overrides:
+      # Don't add random suffix
+      private_key_path: '"test-fixtures/private-key.p8"'
+      team_id: '"9987654321"'
+    test_env_vars:
+      project_id: :PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebase_app_check_device_check_config_full"
+    # Need the time_sleep resource
+    pull_external: true
+    primary_resource_id: "default"
+    vars:
+      bundle_id: "bundle.id.devicecheck"
+      key_id: "Key ID"
+      private_key_path: "path/to/private-key.p8"
+      team_id: "9987654321"
+      token_ttl: "7200s"
+    test_vars_overrides:
+      # Don't add random suffix
+      private_key_path: '"test-fixtures/private-key-2.p8"'
+      team_id: '"9987654321"'
+      token_ttl: '"7200s"'
+    test_env_vars:
+      project_id: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: app_id
+    description: |
+      The ID of an
+      [Apple App](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.iosApps#IosApp.FIELDS.app_id).
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    description: |
+      The relative resource name of the DeviceCheck configuration object
+    output: true
+    pattern: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig
```

Comment:
Even though there are a number of places this is currently set, I don't think this particular property actually does anything. It seems to be legacy. Created https://github.com/hashicorp/terraform-provider-google/issues/17269 to track removing support.

```suggestion
```

Reply:
Done.